code/game/objects/items/mail.dm

```diff
@@ -332,3 +334,166 @@
+		// If the recipient's mind has gone, then anyone can open their mail
+		// whether a mind can actually be qdel'd is an exercise for the reader
+		if(recipient && recipient != user?.mind)
+			to_chat(user, span_notice("You can't open somebody else's mail! That's <em>illegal</em>!"))
+			return
+
+	to_chat(user, span_notice("You start to unwrap the package..."))
+	if(!do_after(user, 1.5 SECONDS, target = user))
+		return
+	user.temporarilyRemoveItemFromInventory(src, TRUE)
+	playsound(loc, 'sound/items/poster_ripped.ogg', 50, TRUE)
+	for(var/obj/item/stuff in contents) // Mail and envelope actually can have more than 1 item.
+		if(user.put_in_hands(stuff) && armed)
+			log_bomber(user, "opened armed mail made by [madeby] ([madeby.ckey]), activating", stuff)
+			stuff.attack_self(user)
+	qdel(src)
+
+/obj/item/mail/traitor/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(armed)
+		if(W.tool_behaviour == TOOL_MULTITOOL)
+			to_chat(user, span_notice("You start to disable something..."))
+			
+			if(do_after(user, 2 SECONDS, target = src))
+				to_chat(user, span_notice("You have disarmed some kind of device..."))
+				playsound(src, 'sound/machines/defib_ready.ogg', 50, TRUE)
+				armed = FALSE
+
+/obj/item/storage/mail_counterfeit_device
+	name = "mail counterfeit device"
+	desc = "Device that actually able to counterfeit NT's mail. This device also able to place a trap inside of mail for malicious actions. Trap will \"activate\" any item inside of mail. Also it might be used for contraband purposes. Integrated micro-computer will give you great configuration optionality for your needs."
+	w_class = WEIGHT_CLASS_NORMAL
+	icon = 'icons/obj/device_syndie.dmi'
+	icon_state = "mail_counterfeit_device" // placeholder for now
+
+/obj/item/storage/mail_counterfeit_device/Initialize(mapload)
+	. = ..()
+	atom_storage.max_slots = 1
+	atom_storage.allow_big_nesting = TRUE
+	atom_storage.max_specific_storage = WEIGHT_CLASS_NORMAL
+
+/obj/item/storage/mail_counterfeit_device/attack_self(mob/user, modifiers)
+	var/mail_type = tgui_alert(user, "Is it gonna be an envelope or a normal mail?", "Mail Counterfeiting", list("Mail", "Envelope"))
+	if(isnull(mail_type))
+		return FALSE
+	if(!(src in user.contents))
+		return FALSE
+	mail_type = lowertext(mail_type)
+
+	var/mail_armed = tgui_alert(user, "Is it gonna be armed?", "Mail Counterfeiting", list("Yes", "No"))
+	if(isnull(mail_armed))
+		return FALSE
+	if(!(src in user.contents))
+		return FALSE
+
+	if(mail_armed == "Yes")
+		mail_armed = TRUE
+	else
+		mail_armed = FALSE
+
+	var/list/mail_recipients = list()
+	var/list/mail_recipients_input_list = list("0# Anyone")
+	var/iterator = 1
+	for(var/mob/living/carbon/human/human in GLOB.human_list)
+		// Skip everyone who is not part of the crew...
+		if(isnull(human.mind))
+			continue
+		if(!(human.mind.assigned_role.job_flags & JOB_CREW_MEMBER))
+			continue
+		mail_recipients += human.mind
+		mail_recipients_input_list += "[iterator]# [human.name]"
```

Comment:
1. Do not include the iterator, it is useless information
2. Sort the list of names before you send it to tgui_input_list

Reply:
Okay, i will sort them, but interator MUST stay. If there will be two members with one name, i will not be able to pick someone specific. Those TGUI lists are kinda... Bad in this way.

Reply:
This doesn't make any sense. If two people have the same name, you're not going to know who is who anyway.

Reply:
i mean... Iterator is here is just to handle duplicates in the easiest way possible. If i will not use the iterator here, i will always get first person with that name. Yea, i can try to make some dupe handler that will scan for dupes and modify dupes names, adding (1) or something else, but that will be additional code that will scan list after every new person added.

Reply:
Two people having the exact same name on the manifest is a once in a thousand round occurrence. We should not sacrifice the common case because of it.

Reply:
And it isn't even relevant  because there's still no way to differentiate which one is your actual target...